// basicprimitivesreact/dist/Templates/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LabelAnnotationTemplate = exports.UserTemplate = exports.ItemTemplate = exports.HighlightTemplate = exports.GroupTitleTemplate = exports.DotHighlightTemplate = exports.CursorTemplate = exports.CheckBoxTemplate = exports.ButtonsTemplate = exports.AnnotationLabelTemplate = void 0;

var _AnnotationLabelTemplate = _interopRequireDefault(require("./AnnotationLabelTemplate"));

var _ButtonsTemplate = _interopRequireDefault(require("./ButtonsTemplate"));

var _CheckBoxTemplate = _interopRequireDefault(require("./CheckBoxTemplate"));

var _CursorTemplate = _interopRequireDefault(require("./CursorTemplate"));

var _DotHighlightTemplate = _interopRequireDefault(require("./DotHighlightTemplate"));

var _GroupTitleTemplate = _interopRequireDefault(require("./GroupTitleTemplate"));

var _HighlightTemplate = _interopRequireDefault(require("./HighlightTemplate"));

var _ItemTemplate = _interopRequireDefault(require("./ItemTemplate"));

var _UserTemplate = _interopRequireDefault(require("./UserTemplate"));

var _LabelAnnotationTemplate = _interopRequireDefault(require("./LabelAnnotationTemplate"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var AnnotationLabelTemplate = _AnnotationLabelTemplate["default"];
exports.AnnotationLabelTemplate = AnnotationLabelTemplate;
var ButtonsTemplate = _ButtonsTemplate["default"];
exports.ButtonsTemplate = ButtonsTemplate;
var CheckBoxTemplate = _CheckBoxTemplate["default"];
exports.CheckBoxTemplate = CheckBoxTemplate;
var CursorTemplate = _CursorTemplate["default"];
exports.CursorTemplate = CursorTemplate;
var DotHighlightTemplate = _DotHighlightTemplate["default"];
exports.DotHighlightTemplate = DotHighlightTemplate;
var GroupTitleTemplate = _GroupTitleTemplate["default"];
exports.GroupTitleTemplate = GroupTitleTemplate;
var HighlightTemplate = _HighlightTemplate["default"];
exports.HighlightTemplate = HighlightTemplate;
var ItemTemplate = _ItemTemplate["default"];
exports.ItemTemplate = ItemTemplate;
var UserTemplate = _UserTemplate["default"];
exports.UserTemplate = UserTemplate;
var LabelAnnotationTemplate = _LabelAnnotationTemplate["default"];
exports.LabelAnnotationTemplate = LabelAnnotationTemplate;
