// basicprimitivesreact/dist/Schemas/ConnectorAnnotationConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _basicprimitives = _interopRequireDefault(require("basicprimitives"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var ConnectorAnnotationConfig = _propTypes["default"].shape({
  annotationType: _propTypes["default"].oneOf([_basicprimitives["default"].common.AnnotationType.Connector]),
  zOrderType: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.ZOrderType)),
  fromItem: _propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].string]),
  toItem: _propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].string]),
  connectorShapeType: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.ConnectorShapeType)),
  connectorPlacementType: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.ConnectorPlacementType)),
  labelPlacementType: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.ConnectorLabelPlacementType)),
  offset: _propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].shape({
    left: _propTypes["default"].number.isRequired,
    top: _propTypes["default"].number.isRequired,
    right: _propTypes["default"].number.isRequired,
    bottom: _propTypes["default"].number.isRequired
  })]),
  lineWidth: _propTypes["default"].number,
  color: _propTypes["default"].string,
  lineType: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.LineType)),
  selectItems: _propTypes["default"].bool,
  label: _propTypes["default"].string,
  size: _propTypes["default"].shape({
    width: _propTypes["default"].number.isRequired,
    height: _propTypes["default"].number.isRequired
  })
});

var _default = ConnectorAnnotationConfig;
exports["default"] = _default;
