// basicprimitivesreact/dist/Schemas/FamDiagramConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _basicprimitives = _interopRequireDefault(require("basicprimitives"));

var _FamDiagramItemConfig = _interopRequireDefault(require("./FamDiagramItemConfig"));

var _TemplateConfig = _interopRequireDefault(require("./TemplateConfig"));

var _BackgroundAnnotationConfig = _interopRequireDefault(require("./BackgroundAnnotationConfig"));

var _ConnectorAnnotationConfig = _interopRequireDefault(require("./ConnectorAnnotationConfig"));

var _HighlightPathAnnotationConfig = _interopRequireDefault(require("./HighlightPathAnnotationConfig"));

var _ShapeAnnotationConfig = _interopRequireDefault(require("./ShapeAnnotationConfig"));

var _LabelAnnotationConfig = _interopRequireDefault(require("./LabelAnnotationConfig"));

var _PaletteItemConfig = _interopRequireDefault(require("./PaletteItemConfig"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Config = _propTypes["default"].shape({
  navigationMode: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.NavigationMode)),
  graphicsType: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.GraphicsType)),
  pageFitMode: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.PageFitMode)),
  minimalVisibility: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.Visibility)),
  orientationType: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.OrientationType)),
  verticalAlignment: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.VerticalAlignmentType)),
  arrowsDirection: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.GroupByType)),
  showExtraArrows: _propTypes["default"].bool,
  extraArrowsMinimumSpace: _propTypes["default"].number,
  groupByType: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.GroupByType)),
  alignBylevels: _propTypes["default"].bool,
  enableMatrixLayout: _propTypes["default"].bool,
  minimumMatrixSize: _propTypes["default"].number,
  maximumColumnsInMatrix: _propTypes["default"].number,
  hideGrandParentsConnectors: _propTypes["default"].bool,
  bevelSize: _propTypes["default"].number,
  elbowType: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.ElbowType)),
  elbowDotSize: _propTypes["default"].number,
  emptyDiagramMessage: _propTypes["default"].string,
  items: _propTypes["default"].arrayOf(_FamDiagramItemConfig["default"]),
  annotations: _propTypes["default"].arrayOf(_propTypes["default"].oneOfType([_BackgroundAnnotationConfig["default"], _ConnectorAnnotationConfig["default"], _HighlightPathAnnotationConfig["default"], _ShapeAnnotationConfig["default"], _LabelAnnotationConfig["default"]])),
  cursorItem: _propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].string]),
  highlightItem: _propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].string]),
  highlightGravityRadius: _propTypes["default"].number,
  selectedItems: _propTypes["default"].arrayOf(_propTypes["default"].oneOfType([_propTypes["default"].number, _propTypes["default"].string])),
  hasSelectorCheckbox: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.Enabled)),
  selectCheckBoxLabel: _propTypes["default"].string,
  selectionPathMode: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.SelectionPathMode)),
  neighboursSelectionMode: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.NeighboursSelectionMode)),
  templates: _propTypes["default"].arrayOf(_TemplateConfig["default"]),
  defaultTemplateName: _propTypes["default"].string,
  defaultLabelAnnotationTemplate: _propTypes["default"].string,
  hasButtons: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.Enabled)),
  onButtonsRender: _propTypes["default"].func,
  normalLevelShift: _propTypes["default"].number,
  dotLevelShift: _propTypes["default"].number,
  lineLevelShift: _propTypes["default"].number,
  normalItemsInterval: _propTypes["default"].number,
  dotItemsInterval: _propTypes["default"].number,
  lineItemsInterval: _propTypes["default"].number,
  cousinsIntervalMultiplier: _propTypes["default"].number,
  itemTitleFirstFontColor: _propTypes["default"].string,
  itemTitleSecondFontColor: _propTypes["default"].string,
  minimizedItemShapeType: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.ShapeType)),
  linesColor: _propTypes["default"].string,
  linesWidth: _propTypes["default"].number,
  linesType: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.LineType)),
  showNeigboursConnectorsHighlighted: _propTypes["default"].bool,
  highlightLinesColor: _propTypes["default"].string,
  highlightLinesWidth: _propTypes["default"].number,
  highlightLinesType: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.LineType)),
  linesPalette: _propTypes["default"].arrayOf(_PaletteItemConfig["default"]),
  calloutMaximumVisibility: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.Visibility)),
  showCallout: _propTypes["default"].bool,
  calloutPlacementOffset: _propTypes["default"].number,
  defaultCalloutTemplateName: _propTypes["default"].string,
  calloutfillColor: _propTypes["default"].string,
  calloutBorderColor: _propTypes["default"].string,
  calloutOffset: _propTypes["default"].number,
  calloutCornerRadius: _propTypes["default"].number,
  calloutPointerWidth: _propTypes["default"].string,
  calloutLineWidth: _propTypes["default"].number,
  calloutOpacity: _propTypes["default"].number,
  buttonsPanelSize: _propTypes["default"].number,
  groupTitlePanelSize: _propTypes["default"].number,
  checkBoxPanelSize: _propTypes["default"].number,
  groupTitlePlacementType: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.AdviserPlacementType)),
  groupTitleOrientation: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].text.TextOrientationType)),
  groupTitleVerticalAlignment: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.VerticalAlignmentType)),
  groupTitleHorizontalAlignment: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.HorizontalAlignmentType)),
  groupTitleFontSize: _propTypes["default"].string,
  groupTitleFontFamily: _propTypes["default"].string,
  groupTitleColor: _propTypes["default"].string,
  groupTitleFontWeight: _propTypes["default"].string,
  groupTitleFontStyle: _propTypes["default"].string,
  distance: _propTypes["default"].number,
  scale: _propTypes["default"].number,
  minimumScale: _propTypes["default"].number,
  maximumScale: _propTypes["default"].number,
  showLabels: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.Enabled)),
  labelSize: _propTypes["default"].shape({
    width: _propTypes["default"].number.isRequired,
    height: _propTypes["default"].number.isRequired
  }),
  labelOffset: _propTypes["default"].number,
  labelOrientation: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].text.TextOrientationType)),
  labelPlacement: _propTypes["default"].oneOf(Object.values(_basicprimitives["default"].common.PlacementType)),
  labelFontSize: _propTypes["default"].string,
  labelFontFamily: _propTypes["default"].string,
  labelColor: _propTypes["default"].string,
  labelFontWeight: _propTypes["default"].string,
  labelFontStyle: _propTypes["default"].string,
  enablePanning: _propTypes["default"].bool,
  autoSizeMinimum: _propTypes["default"].shape({
    width: _propTypes["default"].number.isRequired,
    height: _propTypes["default"].number.isRequired
  }),
  autoSizeMaximum: _propTypes["default"].shape({
    width: _propTypes["default"].number.isRequired,
    height: _propTypes["default"].number.isRequired
  })
});

var _default = Config;
exports["default"] = _default;
